contracts/interchain-token-service/src/contract.rs

```diff
@@ -127,11 +128,13 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .into()
     }
 
-    fn interchain_token_id(env: &Env, sender: Option<Address>, salt: BytesN<32>) -> BytesN<32> {
-        let value = match sender {
-            Some(sender) => (PREFIX_INTERCHAIN_TOKEN_ID, sender, salt).to_xdr(env),
-            None => (salt).to_xdr(env),
+    fn interchain_token_id(env: &Env, sender: Address, salt: BytesN<32>) -> BytesN<32> {
+        let value = if sender == zero_address(env) {
+            (sender, salt).to_xdr(env)
```

Comment:
prefix should always be there, like in EVM

Reply:
ah, i missed it, you're right. updated